src/ldp/http/metadata/BasicMetadataSerializer.ts

```diff
@@ -0,0 +1,26 @@
+import type { HttpResponse } from '../../../server/HttpResponse';
+import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import { MetadataSerializer } from './MetadataSerializer';
+import type { MetadataWriter } from './MetadataWriter';
+
+/**
+ * Basic serializer that calls all its {@link MetadataWriter}s sequentially.
+ */
+export class BasicMetadataSerializer extends MetadataSerializer {
+  private readonly writers: MetadataWriter[];
+
+  public constructor(writers: MetadataWriter[]) {
+    super();
+    this.writers = writers;
+  }
+
+  public async canHandle(): Promise<void> {
+    // Supports all inputs
+  }
```

Comment:
BTW, given how common this case is, we might want to have a base class for `AsyncHandler` with the implementation of an empty body.

Reply:
I don't we really can though. In this case we have `MetadataSerializer extends AsyncHandler<>` and `BasicMetadataSerializer`. Even though `BasicMetadataSerializer` has an empty `canHandle`, this might not be the case for all `MetadataSerializer`s. And there is no (clean) way to inherit from 2 separate classes in Typescript.

Reply:
But `AsyncHandler` is an interface, right?

I'm just suggesting to have a `BaseAsyncHandler` or similar, where `canHandle` is implemented as an empty method. you can use `BaseAsyncHandler` for convenience then, but not necessary in any way.

So in this case, you _could_ have the `abstract class MetadataSerializer` extend from `BaseAsyncHandler` such that it already has a `canHandle` implementation—knowing that you can override it anyway.

That said, now I wonder why `MetadataSerializer` is an abstract class and not a type definition or interface. The base class doesn't do anything?

Reply:
It's actually an abstract class since it has the `handleSafe` function, which is also the reason `MetadataSerializer` and all other handler interfaces are abstract classes.

So could add an empty `canHandle` function to `AsyncHandler`. Would have to remember to override that one when it is necessary then since Webstorm isn't going to add it automatically anymore 😄

Reply:
> So could add an empty `canHandle` function to `AsyncHandler`. Would have to remember to override that one when it is necessary then since Webstorm isn't going to add it automatically anymore 😄

+1 indeed